From: Ganapatrao Kulkarni <ganapatrao.kulkarni@cavium.com>
Date: Mon, 10 Jul 2017 15:24:15 +0800
Subject: ACPICA: iasl: Update to IORT SMMUv3 disassembling
Git-commit: c944230064eb65e4fa018d86779b4fd200b1d7e7
Patch-mainline: v4.14-rc1
References: fate#321947

ACPICA commit 8cadc4fb500e2aa52241e367c87a0f95d9760c58

ARM IORT specification has provision to define Proximity domain
in SMMUv3 IORT table. Adding required changes to decode
Proximity domain of SMMUv3 IORT table.

Link: https://github.com/acpica/acpica/commit/8cadc4fb
Signed-off-by: Ganapatrao Kulkarni <ganapatrao.kulkarni@cavium.com>
Signed-off-by: Bob Moore <robert.moore@intel.com>
Signed-off-by: Lv Zheng <lv.zheng@intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 include/acpi/actbl2.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/acpi/actbl2.h b/include/acpi/actbl2.h
index 5829eedbf3ff..d568948d3a82 100644
--- a/include/acpi/actbl2.h
+++ b/include/acpi/actbl2.h
@@ -806,6 +806,9 @@ struct acpi_iort_smmu_v3 {
 	u32 pri_gsiv;
 	u32 gerr_gsiv;
 	u32 sync_gsiv;
+	u8 pxm;
+	u8 reserved1;
+	u16 reserved2;
 };
 
 /* Values for Model field above */
@@ -818,6 +821,7 @@ struct acpi_iort_smmu_v3 {
 
 #define ACPI_IORT_SMMU_V3_COHACC_OVERRIDE   (1)
 #define ACPI_IORT_SMMU_V3_HTTU_OVERRIDE     (1<<1)
+#define ACPI_IORT_SMMU_V3_PXM_VALID         (1<<3)
 
 /*******************************************************************************
  *
-- 
2.16.1

